src/api_server.py

```diff
@@ -220,9 +221,20 @@ async def chat_stream(request: ChatRequest):
     
     if not _config.model_path:
         raise HTTPException(status_code=500, detail="Model path not configured.")
+
     
     async def event_generator():
         try:
+            # First send the references (page/text pairs)
+            page_nums = get_page_numbers(topk_idxs, _artifacts["meta"])
+            sources_used = []
+            for i, idx in enumerate(topk_idxs):
+                source_text = chunks[i]
+                page = page_nums[idx]
```

Comment:
I think we should handle the case where we do not have page-numbers here (we can may be just ignore such sources, or add them with a placeholder page num). So that it would not lead to a crash for the older index where we do not have page-number in meta.